src/Microsoft.ML.StandardTrainers/Standard/MulticlassClassification/MulticlassNaiveBayesTrainer.cs

```diff
@@ -294,25 +309,33 @@ private NaiveBayesMulticlassModelParameters(IHostEnvironment env, ModelLoadConte
             : base(env, LoaderSignature, ctx)
         {
             // *** Binary format ***
-            // int: _labelCount
-            // int[_labelCount]: _labelHistogram
+            // int: _labelCount (read during reading of _labelHistogram in ReadLongArray())
+            // long[_labelCount]: _labelHistogram
             // int: _featureCount
-            // int[_labelCount][_featureCount]: _featureHistogram
+            // long[_labelCount][_featureCount]: _featureHistogram
             // int[_labelCount]: _absentFeaturesLogProb
-            _labelHistogram = ctx.Reader.ReadIntArray() ?? new int[0];
+            if (ctx.Header.ModelVerWritten >= 0x00010002)
+                _labelHistogram = ctx.Reader.ReadLongArray() ?? new long[0];
+            else
+            {
+                _labelHistogram = Array.ConvertAll(ctx.Reader.ReadIntArray() ?? new int[0], x => (long)x);
+            }
```

Comment:
If ReadIntArray returns null, it likely means the file is bad. Should you be throwing an error in this case?

Reply:
Hey Harish, the array being read from `ctx.Reader.ReadIntArray(int size)` can return null if the size of the array being loaded is 0. Here's the source code:

https://github.com/dotnet/machinelearning/blob/7628d6cd06a0cf8294ddeebbb9613aaec72ab4e7/src/Microsoft.ML.Core/Utilities/Stream.cs#L605-L641